src/PricingManager/Action/CartDiscount.php

```diff
@@ -28,15 +28,30 @@ class CartDiscount implements DiscountInterface, CartActionInterface
 
     protected float $percent = 0;
 
+    /**
+     * @deprecated CartDiscount will reduce the total cart price without affecting added price modifications.
+     */
+    protected bool $onlyDiscountCart = false;
+
     public function executeOnCart(EnvironmentInterface $environment): ActionInterface
     {
         $priceCalculator = $environment->getCart()->getPriceCalculator();
 
+        $subTotal = $priceCalculator->getSubTotal()->getAmount();
         $amount = Decimal::create($this->amount);
-        if ($amount->isZero()) {
-            $amount = $priceCalculator->getSubTotal()->getAmount()->toPercentage($this->getPercent());
-            //round to 2 digits for further calculations to avoid rounding issues at later point
+        if ($subTotal->sub($amount)->isNegative()) {
+            // prevent discounted amount to be higher than the subtotal
+            $amount = $subTotal;
+        } elseif ($amount->isZero()) {
+            $amount = $subTotal->toPercentage($this->getPercent());
+            // round to 2 digits for further calculations to avoid rounding issues at later point
             $amount = Decimal::fromDecimal($amount->withScale(2));
+        } else {
```

Comment:
hmm, on second thought, I wouldn't deprecate the old way. As discussed, there might be use cases for that behavior too.

Reply:
thank you very much! 
I agree for leaving out the percentage-based cart discounts for now